src/app/functions/server/featureFlags.ts

```diff
@@ -34,5 +46,19 @@ export async function getFlag(name: string): Promise<FeatureFlag | undefined> {
     owner: data.owner,
   };
 
-  return flag;
+  if (!flag.isEnabled) {
+    console.warn("Flag is not enabled:", flag.name);
+    return false;
+  }
+
+  if (!flag.allowList?.length) {
+    console.info("Flag does not have an allow list, enabling:", flag.name);
+  }
+
+  if (flag.allowList?.length && !flag.allowList?.includes(session.user.email)) {
+    console.warn("User is not on allow list for flag:", flag.name);
+    return false;
+  }
+
+  return true;
```

Comment:
Minor nit, and would also require changes higher up, but I feel like accidentally enabling a flag might be worse than accidentally disabling it, and so the default might be better as `false`?

```suggestion
  if (!flag.allowList?.length || flag.allowList.includes(session.user.email)) {
    return true;
  }

  return false;
```